rpc/js/src/ClientChannel.ts

```diff
@@ -38,41 +50,32 @@ export class ClientChannel extends BaseChannel {
     dc.addEventListener('close', () => this.onConnectionTerminated());
   }
 
-  public transportFactory(): grpc.TransportFactory {
-    return (opts: grpc.TransportOptions) => {
-      return this.newStream(this.nextStreamID(), opts);
-    };
-  }
-
   private onConnectionTerminated() {
     // we may call this twice but we know closed will be true at this point.
     this.closeWithReason(new ConnectionClosedError('data channel closed'));
-    const err = new ConnectionClosedError('connection terminated');
     for (const streamId in this.streams) {
       const stream = this.streams[streamId]!;
-      stream.cs.closeWithRecvError(err);
+      stream.cs.closeWithRecvError();
     }
   }
 
   private onChannelMessage(event: MessageEvent<any>) {
     let resp: Response;
     try {
-      resp = Response.deserializeBinary(
-        new Uint8Array(event.data as ArrayBuffer)
-      );
+      resp = Response.fromBinary(new Uint8Array(event.data as ArrayBuffer));
     } catch (e) {
       console.error('error deserializing message', e);
       return;
     }
 
-    const stream = resp.getStream();
+    const { stream } = resp;
     if (stream === undefined) {
       console.error('no stream id; discarding');
       return;
     }
 
-    const id = stream.getId();
-    const activeStream = this.streams[id];
+    const { id } = stream;
+    const activeStream = this.streams[Number(id)];
```

Comment:
```suggestion
    const activeStream = this.streams[id];
```

Not sure how likely we are to hit a stream count where this would matter, but maybe we should just use `BigInt` throughout? My understanding is that coercing between `BigInt` and `Number` could lead to precision loss per [MDN docs]( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#description):

> Be careful coercing values back and forth, however, as the precision of a BigInt value may be lost when it is coerced to a Number value.

Reply:
The proto changed this to a bigint so we could also just change the property on the map?

Reply:
> change the property on the map

yep let's do that

Reply:
so bigint cannot be used as an index which makes sense, so how about the record key be a string and we go bigint->string on demand? it has a cost but I don't think we're in that realm of performance needs just yet

Reply:
that works too